src/jdk.incubator.vector/share/classes/jdk/incubator/vector/Float16.java

```diff
@@ -0,0 +1,1667 @@
+        // shift value, in addition to 42, to be used in the
+        // computations below.  Further the (hidden) msb with value 1
+        // in d must be involved as well.
+        int expdelta = 0;
+        long msb = 0x0000_0000_0000_0000L;
+        if (exp < -14) {
+            expdelta = -14 - exp; // FIXME?
+            exp = -15;
+            msb = 0x0010_0000_0000_0000L; // should be 0x0020_... ?
+        }
+        long f_signif_bits = doppel & 0x000f_ffff_ffff_ffffL | msb;
+
+        // Significand bits as if using rounding to zero (truncation).
+        short signif_bits = (short)(f_signif_bits >> (42 + expdelta));
+
+        // For round to nearest even, determining whether or not to
+        // round up (in magnitude) is a function of the least
+        // significant bit (LSB), the next bit position (the round
+        // position), and the sticky bit (whether there are any
+        // nonzero bits in the exact result to the right of the round
+        // digit). An increment occurs in three cases:
+        //
+        // LSB  Round Sticky
+        // 0    1     1
+        // 1    1     0
+        // 1    1     1
+        // See "Computer Arithmetic Algorithms," Koren, Table 4.9
+
+        long lsb    = f_signif_bits & (1L << 42 + expdelta);
+        long round  = f_signif_bits & (1L << 41 + expdelta);
+        long sticky = f_signif_bits & ((1L << 41 + expdelta) - 1);
+
+        if (round != 0 && ((lsb | sticky) != 0 )) {
+            signif_bits++;
+        }
+
+        // No bits set in significand beyond the *first* exponent bit,
+        // not just the significand; quantity is added to the exponent
+        // to implement a carry out from rounding the significand.
+        assert (0xf800 & signif_bits) == 0x0;
+
+        return new Float16((short)(sign_bit | ( ((exp + 15) << 10) + signif_bits ) ));
+    }
+
+    /**
+     * Returns a {@code Float16} holding the floating-point value
+     * represented by the argument string.
+     *
+     * The grammar of strings accepted by this method is the same as
+     * that accepted by {@link Double#valueOf(String)}. The rounding
+     * policy is also analogous to the one used by that method, a
+     * valid input is regarded as an exact numerical value that is
+     * rounded once to the nearest representable {@code Float16} value.
+     *
+     * @apiNote
+     * This method corresponds to the convertFromDecimalCharacter and
+     * convertFromHexCharacter operations defined in IEEE 754.
+     *
+     * @param  s the string to be parsed.
+     * @return the {@code Float16} value represented by the string
+     *         argument.
+     * @throws NullPointerException  if the string is null
+     * @throws NumberFormatException if the string does not contain a
+     *               parsable {@code Float16}.
+     * @see    java.lang.Float#valueOf(String)
+     */
+    public static Float16 valueOf(String s) throws NumberFormatException {
```

Comment:
The current implementation throws when the input is `"NaN"`, with or without an optional sign.

Reply:
Thanks for catching that; fix and tests added in subsequent push.

Reply:
Single digit inputs, e.g., `Float16.valueOf"2")` throw as well.

Reply:
Addressed in most recent push; thanks.